packages/suite/src/workers/export.worker.ts

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable no-restricted-globals */
+import { AccountTransaction } from 'trezor-connect';
+import pdfMake from 'pdfmake/build/pdfmake';
+import pdfFonts from 'pdfmake/build/vfs_fonts';
+import { TDocumentDefinitions } from 'pdfmake/interfaces';
+import { prepareCsv, preparePdf } from '@suite-utils/export';
+import { formatAmount, formatNetworkAmount } from '@wallet-utils/accountUtils';
+import { Network } from '@wallet-types';
+
+pdfMake.vfs = pdfFonts.pdfMake.vfs;
+
+const ctx: Worker = self as any;
+
+type Data = {
+    coin: Network['symbol'];
+    type: 'csv' | 'pdf' | 'json';
+    transactions: AccountTransaction[];
+};
+
+interface CustomMessageEvent extends MessageEvent {
+    data: Data;
+}
+
+const fields = {
+    datetime: 'Date & Time',
+    type: 'Type',
+    txid: 'Transaction ID',
+    addresses: 'Addresses',
+    fee: 'Fee',
+    amount: 'Total',
+};
+
+const addressSeparator = {
+    csv: ' | ',
+    pdf: '\n',
+    json: '',
+};
+
+// Docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/format
+const dateTimeFormat = {
+    hour: 'numeric',
+    minute: 'numeric',
+    second: 'numeric',
+    year: 'numeric',
+    month: 'numeric',
+    day: 'numeric',
+};
+
+const makePdf = (definitions: TDocumentDefinitions): Promise<Blob> =>
+    new Promise(resolve => {
+        pdfMake.createPdf(definitions).getBlob(blob => {
+            resolve(blob);
+        });
+    });
+
+const prepareContent = (data: Data) => {
+    const { coin, type, transactions } = data;
+    return transactions.map(t => {
+        let addresses = [];
+        if (t.tokens.length > 0) {
+            addresses = t.tokens.map(token => {
+                if (token && token.address && token.amount) {
+                    return `${token.address} (${formatAmount(token.amount, token.decimals)} ${
+                        token.symbol
+                    })`;
+                }
+
+                return null;
+            });
+        } else {
+            addresses = t.targets.map(target => {
+                if (target && target.addresses && target.addresses[0] && target.amount) {
```

Comment:
more sugar
`if (target?.addresses?.length && target?.amount) {`

Reply:
Addressed in ca13056.